_posts/2020-03-28-everyone-is-dumb.md

```diff
@@ -0,0 +1,64 @@
+
+After weeks of planning, debate, programming, and refactoring I was nearly finished. I was getting a single persistent bug though, my Item.user was always null (None in Python). I debugged it at the top and bottom (when the item was created and when the user was being referenced), which revealed that somewher in the stack Item.user was being lost. So I discovered every place an Item could be created and refactored them down and made sure all of them set the User correctly.
+
+Still the error persisted.
+
+So I followed all of the references, dropping debug lines all over the place. Every callback, every method, every time the user could have been manipulated or some parameter list might have been borked up, I added some debug printouts.
+
+It all came down to this:
+
+```python
+class Item(object):
+    def __init__(self, game, user, slug):
+
+        logger.debug("user == None is {}", (user is None))
+        logger.debug("game == None is {}", (game is None))
+
+        self.game = game
+        self.user = user
+```
+
+Here my debug lines told me everything was hunky dory, user and game existed and had valid values. What I didn't realize, for another **30 minutes**, was that that two lines below, still in in the init method, was this:
+
+```python
+        self.user = None
+```
+
+Some expletives may have been uttered.
+
+I had left his behind from an earlier idea where the Item's user would be optional. I was so fixated I didn't even bother to look for any other references to the user member until I'd been at it for over 2 hours. I'm not an idiot and I'm not new to this. I _felt_ like an incompetent idiot right then, but 98% of the time I don't make mistakes like this, and I'm willing to be you're the same.
+
+## Days of Present Future
+
+My second story is actually about creating this blog. This site runs on Jekyll using a modified Hacker theme. I had allowed this site to languish, unpublished, in my git repo for 9 months before I finally asked a friend to proof-read my first post and About page. It took me another 4 months after that before I decided to finally pull the trigger. Unfortunately, the two posts stopped appearing on the site when I built it locally.
+
+I knew there were some issues with the Hacker theme, so I looked into it. Turned out to be [a simple issue](https://github.com/pages-themes/hacker/issues/44) of not being compatible with latest major version of Jekyll, which had come out while I let this site gather digital dust. A quick update to the _config.yml fixed that. Except it didn't fix my site, my posts were still not showing up.
+
+Next I went for old faithful, a simple google search. This turned up just a basic list of potential issues, which I was certain couldn't possibly be my problem. But I checked anyway. Lo and behold, my posts were dated for 20**21**-03-18.
+
+![statue facepalming]({{site.url}\assets\images\facepalm.jpg "Facepalm")
```

Comment:
The image should be attributed if possible.

Reply:
Funnily enough, from the "facepalm" article on wikipedia.
Good catch, i'll add the attribution.